core/src/main/java/com/google/errorprone/refaster/UFreeIdent.java

```diff
@@ -99,7 +101,15 @@ public Boolean visitIdentifier(IdentifierTree ident, Void v) {
       if (!isGood) {
         return Choice.none();
       } else if (currentBinding == null) {
+        Symbol currentExprSymbol = ASTHelpers.getSymbol(expression);
+        if (currentExprSymbol instanceof TypeSymbol
+            || (currentExprSymbol instanceof Symbol.MethodSymbol
+                && target instanceof JCTree.JCIdent)) {
```

Comment:
IIUC we can also use an interface here:
```suggestion
                && target instanceof IdentifierTree)) {
```